interfaces/Portalicious/src/app/services/auth/strategies/msal-auth/msal-auth.callback.component.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  inject,
+  OnDestroy,
+} from '@angular/core';
+import { Router } from '@angular/router';
+
+import { MsalService } from '@azure/msal-angular';
+import { AuthenticationResult } from '@azure/msal-browser';
+import { ProgressSpinnerModule } from 'primeng/progressspinner';
+import { Subscription } from 'rxjs';
+
+import { AppRoutes } from '~/app.routes';
+import { AuthService } from '~/services/auth.service';
+import { getReturnUrlFromLocalStorage } from '~/utils/local-storage';
+
+@Component({
+  selector: 'app-msal-auth.callback',
+  standalone: true,
+  imports: [ProgressSpinnerModule],
+  templateUrl: './msal-auth.callback.component.html',
+  styles: ``,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MsalAuthCallbackComponent implements OnDestroy {
+  private authService = inject(AuthService);
+  private router = inject(Router);
+  private msalService = inject(MsalService);
+  private msalSubscription: Subscription | undefined;
+
+  constructor() {
+    this.msalSubscription = this.msalService
+      .handleRedirectObservable()
+      .subscribe((data: AuthenticationResult | null) => {
+        if (!data) {
+          return;
+        }
+        void this.authService
+          .processAzureCallback()
+          .then(async () => {
+            const returnUrl = getReturnUrlFromLocalStorage();
+            if (returnUrl) {
+              await this.router.navigate([returnUrl]);
+            } else {
+              await this.router.navigate(['/', AppRoutes.users]);
+            }
+          })
+          .catch(async () => {
+            await this.router.navigate(['/', AppRoutes.login]);
+          });
```

Comment:
Related to my comment in auth-service -> processAzureCallback

```suggestion
          .catch(async (error) => {
            let errorMessage: string | undefined;

            // TODO: AB#31489 Return a more generic endpoint from the back-end
            if (error instanceof HttpErrorResponse) {
              errorMessage = get(error.error, 'message') as string | undefined;
            }
            if (!errorMessage) {
              errorMessage = $localize`Some generic error message`;
            }

            // somehow pass the error to the login page (I'm not sure how to do this)
            await this.router.navigate(['/', AppRoutes.login, error: errorMessage]);
          });
```